examples/ci-workflowtemplate.yaml

```diff
@@ -0,0 +1,195 @@
+            subPath: GOMODCACHE
+          - mountPath: /root/.cache/go-build
+            name: work
+            subPath: GOCACHE
+        image: golang:1.18
+        workingDir: /go/src/github.com/golang/example
+      outputs:
+        artifacts:
+          - name: GOMODCACHE
+            path: /go/pkg/mod
+            optional: true
+            s3:
+              key: "github.com/golang/examples/{{workflow.parameters.branch}}/GOMODCACHE"
+          - name: GOCACHE
+            path: /root/.cache/go-build
+            optional: true
+            s3:
+              key: "github.com/golang/examples/{{workflow.parameters.branch}}/GOCACHE"
+
+    - name: clone
+      container:
+        volumeMounts:
+          - mountPath: /go/src/github.com/golang/example
+            name: work
+            subPath: src
+          - mountPath: /go/pkg/mod
+            name: work
+            subPath: GOMODCACHE
+          - mountPath: /root/.cache/go-build
+            name: work
+            subPath: GOCACHE
+        image: golang:1.18
+        workingDir: /go/src/github.com/golang/example
+        command: [ sh, -euxc ]
+        args:
+          - |
+            git clone -v -b "{{workflow.parameters.branch}}" --single-branch --depth 1 https://github.com/golang/example.git .
+
+    - name: deps
+      container:
+        image: golang:1.18
+        volumeMounts:
+          - mountPath: /go/src/github.com/golang/example
+            name: work
+            subPath: src
+          - mountPath: /go/pkg/mod
+            name: work
+            subPath: GOMODCACHE
+          - mountPath: /root/.cache/go-build
+            name: work
+            subPath: GOCACHE
+        workingDir: /go/src/github.com/golang/example
+        command: [ sh, -xuce ]
+        args:
+          - |
+            go mod download -x
+
+    - name: build
+      container:
+        image: golang:1.18
+        volumeMounts:
+          - mountPath: /go/src/github.com/golang/example
+            name: work
+            subPath: src
+          - mountPath: /go/pkg/mod
+            name: work
+            subPath: GOMODCACHE
+          - mountPath: /root/.cache/go-build
+            name: work
+            subPath: GOCACHE
+        workingDir: /go/src/github.com/golang/example
+        command: [ sh, -xuce ]
+        args:
+          - |
+            go build ./...
+
+    - name: test
+      container:
+        image: golang:1.18
+        volumeMounts:
+          - mountPath: /go/src/github.com/golang/example
+            name: work
+            subPath: src
+          - mountPath: /go/pkg/mod
+            name: work
+            subPath: GOMODCACHE
+          - mountPath: /root/.cache/go-build
+            name: work
+            subPath: GOCACHE
+        workingDir: /go/src/github.com/golang/example
+        command: [ sh, -euxc ]
+        args:
+          - |
+            go install github.com/jstemmer/go-junit-report@latest
+            go install github.com/alexec/junit2html@v0.0.2
```

Comment:
I had to write this myself.

Reply:
Can we put this in our examples folder as it's a simple script?